data/StateMachine.j2119

```diff
@@ -41,6 +41,8 @@ A Task State MAY have a referencePath field named "TimeoutSecondsPath".
 A Task State MAY have a referencePath field named "HeartbeatSecondsPath".
 A Task State MAY have only one of "TimeoutSeconds" and "TimeoutSecondsPath".
 A Task State MAY have only one of "HeartbeatSeconds" and "HeartbeatSecondsPath".
+A Task State MAY have an object field named "Credentials"; its value is a "TaskCredentials".
+A TaskCredentials MAY have a string field named "roleArn".
```

Comment:
```suggestion
A Task State MAY have an object field named "Credentials".
```



Formally speaking, the Amazon States Language spec doesn't specify the contents of `Credentials` other than that it is a JSON object. Each implementation (Step Functions is just one possible ASL implementation) may have their own structure. Anything to do with ARNs is AWS-specific and wouldn't be included in statelint, which validates the generic ASL spec.

> A Task State MAY include a "Credentials" field, whose value MUST be a JSON object whose value is defined by the interpreter. The States language does not constrain the value of the "Credentials" field. The interpreter will use the specified credentials to execute the work identified by the state's "Resource" field.

https://states-language.net/#task-state

Reply:
I didn't realize that - thanks for the docs link :)